app/src/main/java/com/github/polybooks/database/OLBookDatabase.kt

```diff
@@ -74,6 +54,23 @@ class OLBookDatabase(private val url2json : (String) -> CompletableFuture<JsonEl
             }
     }
 
+    override fun getBooks(isbns: Collection<ISBN>, ordering: BookOrdering): CompletableFuture<List<Book>> {
+        val regularised = isbns.map { regulariseISBN(it) ?: throw IllegalArgumentException("ISBN cannot be regularised") }
+        val futures = regularised.toSet().map { getBook(it) }
+        return listOfFuture2FutureOfList(futures).thenApply { it.filterNotNull() }
+    }
+
+    override fun addBook(book: Book): CompletableFuture<Unit> {
+        //Can't add books to OpenLibrary
+        return CompletableFuture.completedFuture(Unit)
+    }
```

Comment:
throw Unsupported exception?

Reply:
I thought it would break stuff, but as I was writing the explanation of why, I realised that it might not break. So I will probably change this.

Reply:
Well, it does break things. Since OLBookDatabase is as the backend to a CachedBookProvider. If one tries to add a book to it, it breaks.